tests/unit/test_charm.py

```diff
@@ -81,29 +68,27 @@ def read(self, *args, **kwargs):
 ]
 
 
+@patch.object(Container, "restart", new=lambda x, y: True)
 class TestCharm(unittest.TestCase):
     @patch("charm.KubernetesServicePatch", lambda x, y: None)
     def setUp(self):
         self.harness = Harness(GrafanaAgentOperatorCharm)
         self.addCleanup(self.harness.cleanup)
         self.harness.set_model_info(name="lma", uuid="1234567890")
         self.harness.set_leader(True)
-        self.harness.begin()
+        self.harness.begin_with_initial_hooks()
 
     @responses.activate
-    @patch.object(Container, "pull", new=pull_empty_fake_file)
-    @patch.object(Container, "restart")
-    @patch.object(Container, "push")
-    def test_remote_write_configuration(self, mock_push: MagicMock, mock_restart: MagicMock):
-        mock_push.push.return_value = None
-        mock_restart.restart.return_value = True
+    def test_remote_write_configuration(self):
         responses.add(
             responses.POST,
             "http://localhost/-/reload",
             status=200,
         )
 
-        rel_id = self.harness.add_relation("prometheus-remote-write", "prometheus")
+        agent_container = self.harness.charm.unit.get_container("agent")
+
+        rel_id = self.harness.add_relation("send-remote-write", "prometheus")
```

Comment:
This should be called before the first set_leader:
https://github.com/canonical/operator/pull/733

Reply:
That is for peer relations. This is a regular relation.